### PR TITLE
feat: default defmt logging

### PR DESCRIPTION
**Pull request by elenaf9**
# Description

We've agreed that we want to use defmt per default for all logging. This PR now changes all our internal `println!` statements to `log::debug!`, and the examples to `log::info!`.

Concerning the logging level, I think `debug` for all internal stuff makes the most sense, given that they are in fact only relevant for debugging.
The only exception is in `esp_wifi`, where I think `info!` makes sense when we connect/ fail to connect to the wifi. Ideally we'd also have them for `cyw43`, but that's out of scope for this PR.

## Issues/PRs references

Discussed in https://github.com/future-proof-iot/RIOT-rs/pull/349#issuecomment-2214080602.

## Open Questions

- I've enabled the `defmt` module in laze per default for `release`. Do you agree that this makes sense?
- I also wanted to default the `DEFMT_LOG` env to `info` in laze,but somehow it didn't work when I added it as global env to the `defmt` module. Any idea why?

## Change checklist

- [x] I have cleaned up my commit history and squashed fixup commits.
- [x] I have followed the [Coding Conventions](../book/src/coding-conventions.md).
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation. _Will change it if you agree that the defmt module should be selected per default._

### CODE REVIEW

**Comment by kaspar030** [comment]
> * I also wanted to default the `DEFMT_LOG` env to `info` in laze,but somehow it didn't work when I added it as global env to the `defmt` module. Any idea why?

That's because laze "env" does not set the command's shell environment, that needs to be done explicitly. See #355.

File: src/riot-rs-embassy/src/wifi/cyw43.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ pub async fn join(mut control: cyw43::Control<'static>) {
         {
             Ok(_) => break,
             Err(err) => {
-                println!("join failed with status={}", err.status);
+                debug!("join failed with status={}", err.status);
```
[the remark this answers is not in the document]
Comment (elenaf9) [reply src/riot-rs-embassy/src/wifi/cyw43.rs:26]: Discussed out of band; I'll change it to `info!`.

File: laze-project.yml [hunk 2]
```diff
@@ -420,6 +420,7 @@ modules:
   - name: release
     selects:
       - ?lto
+      - ?defmt
```
Comment (kaspar030) [inline laze-project.yml:423]: I think this can actually go into the `riot-rs` context, next to `executor-default`.

**Comment by kaspar030** [comment]
Should we merge #355, then this? I have updated the book chapter with the info from this and #355, should I push that here?

**Comment by elenaf9** [comment]
> Should we merge #355, then this? I have updated the book chapter with the info from this and #355, should I push that here?

Yes sounds good, thanks!

**Comment by kaspar030** [comment]
this needs a rebase now!

**Comment by kaspar030** [comment]
LGTM.

**Pull request merged**